pages/jdl.md

```diff
@@ -23,63 +25,213 @@ The JDL project is [available on GitHub](https://github.com/jhipster/jhipster-co
+If you want to use it in your project, you can add do so by doing:
+  - NPM: `npm install jhipster-core --save`
+  - Yarn: `yarn add jhipster-core`
+  
+to install it locally, and save it in your `package.json` file.
+
+---
 
 ## <a name="jdllanguage"></a> The language
 We tried to keep the syntax as friendly as we can for developers.
-You can do three things with it:
+You can do these things with it:
+  - Declare applications with their options and entities,
   - Declare entities with their attributes,
   - Declare the relationships between them,
   - And declare some JHipster specific options.
+  
+If you wish to view the JDL's grammar, there is an HTML file available
+[here](https://github.com/jhipster/jhipster-core/blob/master/lib/dsl/gen/grammar.html).
+
+
+### <a name="applicationdeclaration"></a> Application declaration
 
+As of v2.0.0, application declaration is possible (compatible with JHipster v5).
+
+_To import one or several applications, you needn't be in a JHipster application folder._
+
+The most basic declaration is done as follows:
+
+```
+application {
+  config {}
+}
+```
+
+A JHipster application has a config with default values and using the previous syntax will
+ensure your application use the default values (as if you didn't make any specific choice).
+The resulting application will have:
+  - baseName: `jhipster`
+  - applicationType: `monolith`
+  - databaseType: `sql`
+  - etc.
+
+Now, if you want some custom options:
+
+```
+application {
+  config {
+    baseName myapp
+    applicationType microservice
+    prodDatabaseType postgresql
+    buildTool gradle
+  }
+}
+```
+
+Those options are only a sample of what's available in the JDL.
+The complete list of options is available in the annexes, [here](#annexes).
+
+If you want more than one application, here's how you do it:
+
+```
+application { // will be generated under the 'myFirstApp' folder
+  config {
+    baseName myFirstApp
+  }
+}
+
+application { // will be generated under the 'mySecondApp' folder
+  config {
+    baseName mySecondApp
+    applicationType microservice
+  }
+}
+```
+
+You can have as many application as you want in as many file as you wish: there's no limitation.
+
+Declaring entities is the most basic thing available and now you can set what entity should be generated
+in the applications you want.
+
+Let's improve the previous example:
+
+```
+application {
+  config {
+    baseName myMonolith
+    applicationType monolith
+  }
+  entities * except C, D
+}
+
+application {
+  config {
+    baseName myGateway
+    applicationType gateway
+    serverPort 9042
+  }
+  entities * except A, B
+}
+
+application {
+  config {
+    baseName microserviceA
+    applicationType microservice
+  }
+  entities C
+}
+
+application {
+  config {
+    baseName microserviceB
+    applicationType microservice
+    serverPort 8082
+  }
+  entities D
+}
+
+entity A
+entity B
+entity C
+entity D
+
+dto * with mapstruct
```

Comment:
This JDL is invalid because `service` is required for the `dto` option.  Fix is to add `service * with serviceClass`

Reply:
Fixed